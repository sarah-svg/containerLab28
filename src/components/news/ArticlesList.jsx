/* eslint-disable react/no-unescaped-entities */
import React from 'react';
import PropTypes from 'prop-types';
import Article from './Article';

const ArticlesList = ({ articles, search }) => {
  const articlesList = articles.map((article, index) => {
    return (
      <li key={index}>
        <Article {...article} />
        <br />
      </li>
    );
  });

  return (
    <>
      {search && (
        <div style={{ textAlign: 'center' }}>
          Fetching results for "{search}"...
        </div>
      )}
      <ul style={{ listStyle: 'none' }}>{articlesList}</ul>
    </>
  );
};

ArticlesList.propTypes = {
  search: PropTypes.string,
  articles: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      author: PropTypes.string,
      description: PropTypes.string,
    }).isRequired
  ),
};

export default ArticlesList;
