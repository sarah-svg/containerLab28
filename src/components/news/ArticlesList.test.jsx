import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ArticlesList from './ArticlesList.jsx';

describe('ArticlesList component', () => {
  afterEach(() => cleanup());
  it('renders ArticlesList', () => {
    const { asFragment } = render(<ArticlesList articles={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
