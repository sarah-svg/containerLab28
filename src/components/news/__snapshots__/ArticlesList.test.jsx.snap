// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesList component renders ArticlesList 1`] = `
<DocumentFragment>
  <ul
    style="list-style: none;"
  />
</DocumentFragment>
`;
